src/base/static/js/views/place-detail-view.js

```diff
@@ -1,23 +1,55 @@
+
   var Util = require('../utils.js');
+  var Gatekeeper = require('../../libs/gatekeeper.js');
 
   var SurveyView = require('mapseed-survey-view');
   var SupportView = require('mapseed-support-view');
+  var RichTextEditorView = require('mapseed-rich-text-editor-view');
 
   var SubmissionCollection = require('../models/submission-collection.js');
 
   module.exports = Backbone.View.extend({
     events: {
       'click .place-story-bar .btn-previous-story-nav': 'onClickStoryPrevious',
-      'click .place-story-bar .btn-next-story-nav': 'onClickStoryNext'
+      'click .place-story-bar .btn-next-story-nav': 'onClickStoryNext',
+      'click #toggle-editor-btn': 'onToggleEditMode',
+      'click #update-place-model-btn': 'onUpdateModel',
+      'click #hide-place-model-btn': 'onHideModel',
+      'click input[data-input-type="binary_toggle"]': 'onBinaryToggle',
+      'change input, textarea': 'saveDraftChanges',
+      'keyup input[name="url-title"]': 'onUpdateUrlTitle'
     },
     initialize: function() {
       var self = this;
 
+      this.isEditable = Util.getAdminStatus(this.options.datasetId);
+      this.isEditingToggled = false;
       this.surveyType = this.options.surveyConfig.submission_type;
       this.supportType = this.options.supportConfig.submission_type;
-      
+      this.isModified = false;
+      this.categoryConfig = _.findWhere(this.options.placeConfig.place_detail, 
+        {category: this.model.get("location_type")});
+      this.commonFormElements = this.options.placeConfig.common_form_elements;
+      this.geometryEditorView = this.options.geometryEditorView;
+      this.onAddAttachmentCallback = null;
+      this.geometryEnabled = (_.find(this.categoryConfig.fields, function(field) {
```

Comment:
I think this could be `_.any(this.categoryConfig.fields, function(field) { return field.type === "geometryToolbar"; })`